renetcode/src/server.rs

```diff
@@ -1,15 +1,32 @@
+use std::fmt::Display;
 use std::{collections::HashMap, net::SocketAddr, time::Duration};
 
 use crate::{
     crypto::generate_random_bytes,
     packet::{ChallengeToken, Packet},
     replay_protection::ReplayProtection,
     token::PrivateConnectToken,
-    ClientID, NetcodeError, NETCODE_CONNECT_TOKEN_PRIVATE_BYTES, NETCODE_CONNECT_TOKEN_XNONCE_BYTES, NETCODE_KEY_BYTES, NETCODE_MAC_BYTES,
+    NetcodeError, NETCODE_CONNECT_TOKEN_PRIVATE_BYTES, NETCODE_CONNECT_TOKEN_XNONCE_BYTES, NETCODE_KEY_BYTES, NETCODE_MAC_BYTES,
     NETCODE_MAX_CLIENTS, NETCODE_MAX_PACKET_BYTES, NETCODE_MAX_PAYLOAD_BYTES, NETCODE_MAX_PENDING_CLIENTS, NETCODE_SEND_RATE,
     NETCODE_USER_DATA_BYTES, NETCODE_VERSION_INFO,
 };
 
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub struct NetworkId(pub u64);
```

Comment:
The NetworkId should not be in the `renetcode` crate. This crate can be disabled so we need to move this struct to the renet crate. We can keep using an u64 here, most people will not use this crate directly anyway.

Reply:
```suggestion
pub struct ClientId(u64);
```

I think this is more intuitive.